RecoLocalFastTime/FTLCommonAlgos/plugins/MTDRecHitAlgo.cc

```diff
@@ -0,0 +1,75 @@
+#include "RecoLocalFastTime/FTLCommonAlgos/interface/MTDRecHitAlgoBase.h"
+
+class MTDRecHitAlgo : public MTDRecHitAlgoBase {
+ public:
+  /// Constructor
+  MTDRecHitAlgo( const edm::ParameterSet& conf,
+                       edm::ConsumesCollector& sumes ) : 
+    MTDRecHitAlgoBase( conf, sumes ),
+    BTLthresholdToKeep_( conf.getParameter<double>("BTLthresholdToKeep") ),
+    BTLcalibration_( conf.getParameter<double>("BTLcalibrationConstant") ),
+    ETLthresholdToKeep_( conf.getParameter<double>("ETLthresholdToKeep") ),
+    ETLcalibration_( conf.getParameter<double>("ETLcalibrationConstant") ) { }
+
+  /// Destructor
+  ~MTDRecHitAlgo() override { }
+
+  /// get event and eventsetup information
+  void getEvent(const edm::Event&) final {}
+  void getEventSetup(const edm::EventSetup&) final {}
+
+  /// make the rec hit
+  FTLRecHit makeBTLRecHit(const FTLUncalibratedRecHit& uRecHit, uint32_t& flags ) const final;
+  FTLRecHit makeETLRecHit(const FTLUncalibratedRecHit& uRecHit, uint32_t& flags ) const final;
+
+ private:  
+  double BTLthresholdToKeep_, BTLcalibration_;
+  double ETLthresholdToKeep_, ETLcalibration_;
+};
+
+FTLRecHit 
+MTDRecHitAlgo::makeBTLRecHit(const FTLUncalibratedRecHit& uRecHit, uint32_t& flags ) const { 
```

Comment:
The body of this function and the next one are basically the same. To reduce code duplication, these functions could both call a helper function, passing in the calibration and threshold values for BTL or ETL.

Reply:
Removed the duplication.